src/authorizations/actions/thunks.ts

```diff
@@ -168,6 +170,35 @@ export const deleteAuthorization = (id: string, name: string = '') => async (
   }
 }
 
+export const bulkDeleteAuthorizations = (tokenIds: string[]) => async (
+  dispatch: Dispatch<Action | NotificationAction>
+) => {
+  const deletePromises = tokenIds.map(async (tokenId: string) => {
+    const resp = await api.deleteAuthorization({authID: tokenId})
+
+    if (resp.status !== 204) {
+      return Promise.reject(new Error(resp.data.message))
+    }
+    dispatch(removeAuthorization(tokenId))
```

Comment:
since we're already mapping through and creating a promise, it would be great to optimize this to reduce the number of dispatches when this is called from N to 1. We can do this by simply returning the `tokenId` here, then dispatching the `results` below (i.e. all the tokenIds that have been successfully deleted and all the promises settled). This would require a new action to `removeAuthorizations` (plural) that basically does the same thing a the `removeAuthorization`, except for an array of ids rather than 1 id. The benefit here is that we reduce the number of render cycles and make this cool new feature more reactive.

Reply:
I don't think it's necessary to change redux to do a batch operation at this point. Users performing an operation on a list of items are going to have different expectations of the page's behavior than when they navigate to a page displaying resources. This is an experimental feature that we're testing the waters on (and trying to get out quickly and with minimal changes) and I don't want to add redux operations until we're sure this will be used throughout the app and we have a better idea of how the bulk actions should behave.

Technically this function does return a promise, but it doesn't include all the request information the original does, just the resolution status of `resolved` or `rejected`. Academically I think being explicit and returning the promise is the 'correct' course of action, but pramgatically _as it stands now_, it doesn't matter. The value isn't being used by the consuming function (`Promise.allSettled`'s function) and since the thunk dispatches a redux evenut and the UI doesn't use the promise, it's not necessary.

I think it's a best practice to return `resp` so that if we change how it's used in the future, we're not left scratching our heads wondering why it's not working the way we expect.

Reply:
Hmmm, ok. If you’re ok with that then I’m not gonna block it. If this is experimental, should we be feature flagging it? Also, we should we write up an issue to address optimizations so that we don’t end up scratching our heads if the bulk actions ever grows in scope and we’re left with performance issues?

Reply:
We already have the feature flag! `bulkActionDeleteTokens` used in the PR! I can write up an issue for the future!